sentry-android-core/src/main/java/io/sentry/android/core/ActivityLifecycleIntegration.java

```diff
@@ -0,0 +1,181 @@
+      application.registerActivityLifecycleCallbacks(this);
+      options.getLogger().log(SentryLevel.DEBUG, "ActivityBreadcrumbsIntegration installed.");
+    }
+  }
+
+  private boolean isPerformanceEnabled(final @NotNull SentryAndroidOptions options) {
+    return ((options.getTracesSampleRate() != null || options.getTracesSampler() != null)
+        && options.isEnableAutoActivityLifecycleTracing());
+  }
+
+  @Override
+  public void close() throws IOException {
+    application.unregisterActivityLifecycleCallbacks(this);
+
+    options.getLogger().log(SentryLevel.DEBUG, "ActivityBreadcrumbsIntegration removed.");
+  }
+
+  private void addBreadcrumb(final @NonNull Activity activity, final @NotNull String state) {
+    if (options.isEnableActivityLifecycleBreadcrumbs()) {
+      final Breadcrumb breadcrumb = new Breadcrumb();
+      breadcrumb.setType("navigation");
+      breadcrumb.setData("state", state);
+      breadcrumb.setData("screen", getActivityName(activity));
+      breadcrumb.setCategory("ui.lifecycle");
+      breadcrumb.setLevel(SentryLevel.INFO);
+      hub.addBreadcrumb(breadcrumb);
+    }
+  }
+
+  private @NotNull String getActivityName(final @NonNull Activity activity) {
+    return activity.getClass().getSimpleName();
+  }
+
+  private void startTracing(final @NonNull Activity activity) {
+    if (performanceEnabled && !isRunningTransaction(activity)) {
+      final ITransaction transaction =
+          hub.startTransaction(getActivityName(activity), "navigation");
+
+      // lets bind to the scope so other integrations can pick it up
+      hub.configureScope(
+          scope -> {
+            // we we'd not like to overwrite existent transactions bound to the Scope manually.
+            if (scope.getTransaction() == null) {
+              scope.setTransaction(transaction);
+            }
+          });
+
+      activities.put(activity, transaction);
+    }
+  }
+
+  private boolean isRunningTransaction(final @NonNull Activity activity) {
+    return activities.containsKey(activity);
+  }
+
+  private void stopTracing(final @NonNull Activity activity) {
+    if (performanceEnabled && options.isEnableAutoActivityLifecycleTracingFinish()) {
+      final ITransaction transaction = activities.get(activity);
+      if (transaction != null) {
+        SpanStatus status = transaction.getStatus();
+        // status might be set by other integrations, let's not overwrite it
+        if (status == null) {
+          status = SpanStatus.OK;
+        }
+        transaction.finish(status);
+      }
+    }
+  }
+
+  @Override
+  public synchronized void onActivityCreated(
+      @NonNull Activity activity, @Nullable Bundle savedInstanceState) {
+    addBreadcrumb(activity, "created");
+
+    // if activity has global fields being init. and
+    // they are slow, this won't count the whole fields/ctor initialization time, but only
+    // when onCreate is actually called.
+    startTracing(activity);
```

Comment:
`m`: Why not call this in `onActivityPreCreated`?

Reply:
this is called already `before` the App's onCreate, so its not necessary.

Reply:
I don't think so. When taking a look at the source code of Activity, there is a lot of stuff happening before `onActivityCreated` is called, like restoring fragments, stuff with `savedInstanceState`, etc. This could also be slow and I think it would be good to include that.

Reply:
the user's code is only executed after this method, not before, except the global fields as stated before.

Reply:
added `onActivityPreCreated` for thr sake of correctness